File: backend/app/models/committee_member.py
from sqlalchemy import Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base, UUIDPrimaryKey


class CommitteeMember(UUIDPrimaryKey, Base):
    __tablename__ = "committee_members"

    name: Mapped[str] = mapped_column(String(200), nullable=False)
    position: Mapped[str] = mapped_column(String(150), nullable=False)
    academic_year: Mapped[str] = mapped_column(String(20), nullable=False, index=True)
    photo_url: Mapped[str | None] = mapped_column(String(500))
    linkedin_url: Mapped[str | None] = mapped_column(String(500))
    sort_order: Mapped[int] = mapped_column(Integer, nullable=False, default=0, server_default="0")

File: client_app/lib/models/committee_member.dart
class CommitteeMember {
  CommitteeMember({
    required this.id,
    required this.name,
    required this.position,
    required this.academicYear,
    required this.photoUrl,
    required this.linkedinUrl,
    required this.sortOrder,
  });

  final String id;
  final String name;
  final String position;
  final String academicYear;
  final String? photoUrl;
  final String? linkedinUrl;
  final int sortOrder;

  factory CommitteeMember.fromJson(Map<String, dynamic> json) {
    return CommitteeMember(
      id: json['id'] as String,
      name: json['name'] as String,
      position: json['position'] as String,
      academicYear: json['academic_year'] as String,
      photoUrl: json['photo_url'] as String?,
      linkedinUrl: json['linkedin_url'] as String?,
      sortOrder: json['sort_order'] as int,
    );
  }
}
